swap_meet/vendor.py

```diff
@@ -1,2 +1,105 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        # if no inventory, default to empty list
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        # always add item to inventory
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        # remove item if it exists
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+
+    # part of wave 2
+    def get_by_id(self, id):
+        # find item in inventory by its unique id
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+
+
+    # wave 3
+    # removes my_item from this Vendor inventory and adds to friend inventory
+    # removes their_item from other Vendor inventory and adds it to this inventory
+    # method returns True
+    # If vendor inventories do not contain my_item or their_item
+    # return False
+    def swap_items(self, other_vendor, my_item, their_item):
+        self.other_vendor = other_vendor
+        self.my_item = my_item
+        self.their_item = their_item
+        
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item)
+            other_vendor.inventory.append(my_item)
+            other_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
```

Comment:
No need to do this check since line 39 just checked if the items were not in inventory. If the items are checked and and the condition on line 39 is not true, then we can assume that the items are in the inventory.

Then we can pull all this logic out of this if-block an unindent it. 

```suggestion
        self.inventory.remove(my_item)
        other_vendor.inventory.append(my_item)
        other_vendor.inventory.remove(their_item)
        self.inventory.append(their_item)
        return True
```